### ISSUE TITLE
[serialization] Value stored to 'length' is never read (Trac #1785)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1777]
a whole bunch of static analysis bugs reported for each assignment of `length` from the copy pasta in 
`serialization/private/serialization/archive_exception.cpp`

for example see http://software.icecube.wisc.edu/static_analysis/2016-07-21-030215-62891-1/report-09d2ea.html#EndPath
full list can be found at http://software.icecube.wisc.edu/static_analysis/2016-07-21-030215-62891-1/

Migrated from https://code.icecube.wisc.edu/ticket/1785
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T23:22:40", 
    "description": "a whole bunch of static analysis bugs reported for each assignment of `length` from the copy pasta in \n`serialization/private/serialization/archive_exception.cpp`\n\nfor example see http://software.icecube.wisc.edu/static_analysis/2016-07-21-030215-62891-1/report-09d2ea.html#EndPath\nfull list can be found at http://software.icecube.wisc.edu/static_analysis/2016-07-21-030215-62891-1/", 
    "reporter": "kjmeagher", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547248960560945", 
    "component": "cmake", 
    "summary": "[serialization] Value stored to 'length' is never read", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-07-21T17:33:47", 
    "milestone": "", 
    "owner": "cweaver", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160721T20:15:06: 
* cweaver changed status from "new" to "closed"
* cweaver changed resolution from "" to "fixed"
* cweaver commented:

> In changeset 2625:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2625"
> The length variable is used on some paths, and keeping the same code structure on all paths is a good way to avoid forgetting to update it. 
> Tell the static analyzer that this is a deliberate choice, and not worth reporting. 
> Closes #1777.
> ```


* cweaver changed owner from "" to "cweaver"

**Comment by icecube-trac** [comment]
Trac update at 20190111T23:22:40: nwhitehorn commented:

> In changeset 169957:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="169957"
> The length variable is used on some paths, and keeping the same code structure on all paths is a good way to avoid forgetting to update it. 
> Tell the static analyzer that this is a deliberate choice, and not worth reporting. 
> Closes #1777.
> ```
